pynamodb/models.py

```diff
@@ -267,6 +264,19 @@ def __init__(self, name, bases, namespace, discriminator=None) -> None:
                 }
                 cls.DoesNotExist = type('DoesNotExist', (DoesNotExist, ), exception_attrs)
 
+    @staticmethod
+    def _initialize_indexes(cls):
+        """
+        Initialize indexes on the class.
+        """
+        cls._indexes = {}
+        for name, index in getmembers(cls, lambda o: isinstance(o, Index)):
+            if not hasattr(index.Meta, "model"):
+                index.Meta.model = cls
```

Comment:
This changes the current behavior where the class is set by any model that has the index in the class namespace. Now first write wins.